apex/contrib/group_norm/group_norm.py

```diff
@@ -40,86 +39,115 @@ def torch_group_norm(x, g, w, b, eps, act=""):
         y = y.to(dtype=xdtype)
     return y
 
-
-class GroupNormNHWC(torch.autograd.Function):
-
-    @staticmethod
-    @disable  # This shouldn't be captured by TorchDynamo
-    def forward(ctx,
-                x,
-                G,
-                weight,
-                bias,
-                eps,
-                act="",
-                algo=group_norm_cuda.OnePass):
-        # sanity check
-        act = act.lower()
-        assert x.is_contiguous(memory_format=torch.channels_last), \
-            "Only support NHWC layout."
-        assert weight.numel() == x.shape[1], "Unexpected parameter count."
-        assert bias.numel() == x.shape[1], "Unexpected parameter count."
-        assert x.shape[1] % G == 0, "C % G != 0."
-        assert act in ["", "silu", "swish"], "Unsupported activation."
-
-        with_swish = (act in ["silu", "swish"])
-
-        # enqueue fprop kernel
-        y, sums = group_norm_cuda.forward(x, G, weight, bias, eps, algo,
+@torch.library.custom_op("apex::group_norm_nhwc_fprop", mutates_args=())
+def group_norm_nhwc_fprop(
+    x: torch.Tensor,
+    G: int,
+    weight: torch.Tensor,
+    bias: torch.Tensor,
+    eps: float,
+    act: str | None=None,
+    passes: int=1,
+) -> tuple[torch.Tensor, torch.Tensor]:
+    # sanity check
+    act = act.lower() if act else act
+    assert x.is_contiguous(memory_format=torch.channels_last), \
+        "Only support NHWC layout."
+    assert weight.numel() == x.shape[1], "Unexpected parameter count."
+    assert bias.numel() == x.shape[1], "Unexpected parameter count."
+    assert x.shape[1] % G == 0, "C % G != 0."
+    assert act in [None, "", "silu", "swish"], "Unsupported activation."
+    assert passes in [1, 2], "Invalid number of passes for algorithm."
+
+    with_swish = (act in ["silu", "swish"])
```

Comment:
nit-picking, cosmetic

```suggestion
    with_swish = act in ("silu", "swish")
```